File: Android/Kotlin/DogListDemo/app/src/main/java/com/suavepirate/doglistdemo/MainActivity.kt
package com.suavepirate.doglistdemo

import android.os.Bundle
import android.support.design.widget.Snackbar
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager
import android.view.Menu
import android.view.MenuItem

import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.content_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbar)

        fab.setOnClickListener { view ->
            Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
                .setAction("Action", null).show()
        }

        dogRecyclerView.layoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
        dogRecyclerView.adapter = DogListAdapter(listOf(
             AdoptableDog("Bentley", 5, true),
             AdoptableDog("Ryder", 2, true),
             AdoptableDog("Belle", 7, true),
             AdoptableDog("Tootsie", 5, true),
             AdoptableDog("Smokey", 5, true),
             AdoptableDog("Biscuit", 1, true),
             AdoptableDog("Jazzy", 15, true),
             AdoptableDog("Hunter", 3, false)
        ), this)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        return when (item.itemId) {
            R.id.action_settings -> true
            else -> super.onOptionsItemSelected(item)
        }
    }
}

File: Android/CSharp/DogListDemo/MainActivity.cs
using System;
using Android.App;
using Android.OS;
using Android.Runtime;
using Android.Support.Design.Widget;
using Android.Support.V7.App;
using Android.Support.V7.Widget;
using Android.Views;
using Android.Widget;
using DogListDemo.Models;

namespace DogListDemo
{
    [Activity(Label = "@string/app_name", Theme = "@style/AppTheme.NoActionBar", MainLauncher = true)]
    public class MainActivity : AppCompatActivity
    {
        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);
            SetContentView(Resource.Layout.activity_main);

            Android.Support.V7.Widget.Toolbar toolbar = FindViewById<Android.Support.V7.Widget.Toolbar>(Resource.Id.toolbar);
            SetSupportActionBar(toolbar);
            var fab = FindViewById<FloatingActionButton>(Resource.Id.fab);
            fab.Click += FabOnClick;

            var recycler = FindViewById<RecyclerView>(Resource.Id.DogRecyclerView);
            var dogs = new AdoptableDog[] {
                new AdoptableDog("Bentley", 5, true),
                new AdoptableDog("Ryder", 2, true),
                new AdoptableDog("Belle", 7, true),
                new AdoptableDog("Tootsie", 5, true),
                new AdoptableDog("Smokey", 5, true),
                new AdoptableDog("Biscuit", 1, true),
                new AdoptableDog("Jazzy", 15, true),
                new AdoptableDog("Hunter", 3, false)
            };
            recycler.SetLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.Vertical, false));
            recycler.SetAdapter(new DogListAdapter(dogs, this));
        }

        public override bool OnCreateOptionsMenu(IMenu menu)
        {
            MenuInflater.Inflate(Resource.Menu.menu_main, menu);
            return true;
        }

        public override bool OnOptionsItemSelected(IMenuItem item)
        {
            int id = item.ItemId;
            if (id == Resource.Id.action_settings)
            {
                return true;
            }

            return base.OnOptionsItemSelected(item);
        }

        private void FabOnClick(object sender, EventArgs eventArgs)
        {
            var view = (View) sender;
            Snackbar.Make(view, "Replace with your own action", Snackbar.LengthLong)
                .SetAction("Action", (Android.Views.View.IOnClickListener)null).Show();
        }
	}
}

